client/vscode/src/api.ts

```diff
@@ -10,15 +18,25 @@ export interface ExtensionApi {
      * If this API changes, the version number will be incremented.
      */
     apiVersion(version: 1): {
+        /**
+         * Get OpenCtx provider capabilities.
+         */
+        capabilities(params: CapabilitiesParams): Promise<CapabilitiesResult[] | null>
+
+        /**
+         * Get OpenCtx candidate items for the document.
+         */
+        candidateItems(params: CandidateItemsParams): Promise<CandidateItemsResult | null>
+
         /**
          * Get OpenCtx items for the document.
          */
-        getItems(params: ItemsParams): Promise<Item[] | null>
+        items(params: ItemsParams): Promise<ItemsResult | null>
```

Comment:
side-note: what is the motivation for all of these methods optionally returning null?

Reply:
no good reason. I guess `null` could mean "I knew I wouldn't have any results before I tried" and `[]` could mean "I might've had results but nah" but that is not clear or meaningful necessarily. So we can just remove a null option.